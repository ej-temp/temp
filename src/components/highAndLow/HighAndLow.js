import React from 'react';
import PropTypes from 'prop-types';

export default function HighAndLowTemp({ lowTemp, highTemp, avgTemp }) {
  return (
    <>
      <h2>low</h2>
      <h4>location: {lowTemp.name}</h4>
      <h5>Temperature: {lowTemp.temp}</h5>
      <h2>high</h2>
      <h4>location: {highTemp.name}</h4>
      <h5>Temperature: {highTemp.temp}</h5>
      <h2>avg</h2>
      <h5>Temperature: {avgTemp}</h5>
    </>
  );
}

HighAndLowTemp.propTypes = {
  lowTemp: PropTypes.shape({
    name: PropTypes.string.isRequired,
    temp: PropTypes.number
  }).isRequired,
  highTemp: PropTypes.shape({
    name: PropTypes.string.isRequired,
    temp: PropTypes.number
  }).isRequired,
  avgTemp: PropTypes.number
};
