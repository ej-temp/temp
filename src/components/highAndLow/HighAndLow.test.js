import React from 'react';
import { shallow } from 'enzyme';
import HighAndLow from './HighAndLow';

describe('HighAndLow component', () => {
  it('renders HighAndLow', () => {
    const wrapper = shallow(<HighAndLow 
      lowTemp={{ name: '', temp: 20 }}
      highTemp={{ name: '', temp: 20 }}
      avgTemp={5}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
