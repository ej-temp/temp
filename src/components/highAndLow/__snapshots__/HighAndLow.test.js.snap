// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighAndLow component renders HighAndLow 1`] = `
<Fragment>
  <h2>
    low
  </h2>
  <h4>
    location: 
  </h4>
  <h5>
    Temperature: 
    20
  </h5>
  <h2>
    high
  </h2>
  <h4>
    location: 
  </h4>
  <h5>
    Temperature: 
    20
  </h5>
  <h2>
    avg
  </h2>
  <h5>
    Temperature: 
    5
  </h5>
</Fragment>
`;
